contributors/design-proposals/dynamic-admission-control-configuration.md

```diff
@@ -0,0 +1,332 @@
+	// the timeout is reached; If "Fail" is set, apiserver returns timeout error
+	// if the timeout is reached.
+	FailurePolicy *FailurePolicyType `json:"failurePolicy,omitempty" protobuf:"bytes,2,opt,name=failurePolicy"`
+}
+
+type FailurePolicyType string
+
+const (
+	// Ignore means the initilizer is removed from the initializers list of an
+	// object if the initializer is timed out.
+	Ignore FailurePolicyType = "Ignore"
+	// For 1.7, only "Ignore" is allowed. "Fail" will be allowed when the
+	// extensible admission feature is beta.
+	Fail FailurePolicyType = "Fail"
+)
+
+// ExternalAdmissionHook describes an external admission webhook and the
+// resources and operations it applies to.
+type ExternalAdmissionHook struct {
+	// The name of the external admission webhook.
+	// Required.
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+
+	// ClientConfig defines how to communicate with the hook.
+	// Required
+	ClientConfig AdmissionHookClientConfig `json:"clientConfig" protobuf:"bytes,2,opt,name=clientConfig"`
+
+	// Rules describes what operations on what resources/subresources the webhook cares about.
+	// The webhook cares about an operation if it matches _any_ Rule.
+	Rules []Rule `json:"rules,omitempty" protobuf:"bytes,3,rep,name=rules"`
+
+	// FailurePolicy defines how unrecognized errors from the admission endpoint are handled -
+	// allowed values are Ignore or Fail. Defaults to Ignore.
+	// +optional
+	FailurePolicy *FailurePolicyType
+}
+
+// Rule describes the Verbs and Resources an admission hook cares about. Each
+// Rule is a tuple of Verbs and Resources.It is recommended to make sure all
+// the tuple expansions are valid.
+type Rule struct {
+	// Verbs is the verbs the admission hook cares about - CREATE, UPDATE, or *
+	// for all verbs.
+	// If '*' is present, the length of the slice must be one.
+	// Required.
+	Verbs []OperationType `json:"verbs,omitempty" protobuf:"bytes,1,rep,name=verbs"`
+
+	// APIGroups is the API groups the resources belong to. '*' is all groups.
+	// If '*' is present, the length of the slice must be one.
+	// Required.
+	APIGroups []string `json:"apiGroups,omitempty" protobuf:"bytes,2,rep,name=apiGroups"`
+
+	// APIVersions is the API versions the resources belong to. '*' is all versions.
+	// If '*' is present, the length of the slice must be one.
+	// Required.
+	APIVersions []string `json:"apiVersions,omitempty" protobuf:"bytes,3,rep,name=apiVersions"`
+
+	// Resources is a list of resources this rule applies to.
+	//
+	// For example:
+	// 'pods' means pods.
+	// 'pods/log' means the log subresource of pods.
+	// '*' means all resources, but not subresources.
+	// 'pods/*' means all subresources of pods.
+	// '*/scale' means all scale subresources.
+	// '*/*' means all resources and their subresources.
+	//
+	// If '*' or '*/*' is present, the length of the slice must be one.
+	// Required.
+	Resources []string `json:"resources,omitempty" protobuf:"bytes,4,rep,name=resources"`
```

Comment:
Simple in one case means harder or impossible for another.  The idea that you can have a rule dictate whether matches get processed _(sent to the webhook)_ or not is a powerful thing.  The whole idea of the declarative filtering/matching was to make it super clear what things you were interested in processing in your webhook and those you aren't.  As you attempt to simplify things, you negatively impact performance because if we can't be finer grained about what to send over the wire, we risk sending too much over the wire.  That was my thought.